### PR TITLE
Single-character macros like active characters

### PR DESCRIPTION
**Pull request by edemaine**
This tiny PR adds support for redefining single characters by arbitrary macros, while also simplifying the code slightly.  I don't see a reason not to allow this, and it would be immediately useful for resolving #967, as well as e.g. users adding support for Unicode characters that KaTeX doesn't already support, or which KaTeX already supports but the user wants to override in a different way.  Overall seems like a win.

### CODE REVIEW

**Comment by ronkok** [comment]
Cool. This will be helpful for double-struck Unicode characters like ℂ or ℕ.

**Comment by edemaine** [comment]
@ronkok Great! More motivation for this change.

File: README.md [hunk 1]
```diff
@@ -48,7 +48,7 @@ You can provide an object of options as the last argument to `katex.render` and
...
 - `throwOnError`: `boolean`. If `true`, KaTeX will throw a `ParseError` when it encounters an unsupported command. If `false`, KaTeX will render the unsupported command as text in the color given by `errorColor`. (default: `true`)
 - `errorColor`: `string`. A color string given in the format `"#XXX"` or `"#XXXXXX"`. This option determines the color which unsupported commands are rendered in. (default: `#cc0000`)
-- `macros`: `object`. A collection of custom macros. Each macro is a property with a name like `\name` (written `"\\name"` in JavaScript) which maps to a string that describes the expansion of the macro.
+- `macros`: `object`. A collection of custom macros. Each macro is a property with a name like `\name` (written `"\\name"` in JavaScript) which maps to a string that describes the expansion of the macro. The behavior of single characters can also be redefined by a macro (similar to TeX active characters).
```
Comment (kevinbarabash) [inline README.md]: The tests allow single characters for the names, but reading this it sounds like only things like `\name` are accepted.  Maybe change the last sentence to read like: 
```
Single characters keys can also be included in which case the character will be redefined as the given macro (similar to TeX active characters).
```

File: test/katex-spec.js [hunk 2]
```diff
@@ -2603,6 +2603,12 @@ describe("A macro expander", function() {
         expect("X \\implies Y").toBuild();
         expect("X \\impliedby Y").toBuild();
     });
+
+    it("should allow aliasing characters", function() {
+        compareParseTree("x’=c", "x'=c", {
+            "’": "'",
+        });
```
Comment (kevinbarabash) [inline test/katex-spec.js]: Nice test.

File: README.md (hunk 1 above)
Comment (edemaine) [reply README.md]: Good idea!  Revised.

**Comment by kevinbarabash** [comment]
@edemaine thanks for make this change.  It's really going to help with the unicode effort.

**Pull request merged**